src/lib/languageUtils.ts

```diff
@@ -62,29 +62,36 @@ export function getVariableAssignmentRegex(
   }
 }
 
-export function checkGateReplacement(language: SupportedLanguageType): string {
+export function checkGateReplacement(
+  language: SupportedLanguageType,
+  value?: unknown,
```

Comment:
should this be a boolean?

Reply:
We expect this to be a boolean, but I left it untyped to give more flexibility in the default value we can use. But on second thought, we should just type gates to `boolean` and configs to `Record`